ngraph/test/type_prop/unary_base.cpp

```diff
@@ -0,0 +1,178 @@
+        test_name.erase(test_name.end() - 1);
+        return test_name;
+    }
+
+    void SetUp()
+    {
+        const auto& type = std::get<0>(GetParam());
+        const auto& shape = std::get<1>(GetParam());
+
+        input_type = type;
+        input_shape = shape;
+    }
+};
+
+class IncompatibleInputType : public testing::WithParamInterface<UnaryParams>, public testing::Test
+{
+public:
+    InputType input_type;
+    Shape input_shape;
+
+    static std::string getTestCaseName(testing::TestParamInfo<UnaryParams> obj)
+    {
+        InputType type;
+        Shape shape;
+        std::ostringstream result;
+        std::tie(type, shape) = obj.param;
+        result << type << "_";
+        result << shape;
+        auto test_name = result.str();
+        std::replace_if(
+            test_name.begin(), test_name.end(), [](char c) { return !std::isalnum(c); }, '_');
+        test_name.erase(test_name.end() - 1);
+        return test_name;
+    }
+
+    void SetUp()
+    {
+        const auto& type = std::get<0>(GetParam());
+        const auto& shape = std::get<1>(GetParam());
+
+        input_type = type;
+        input_shape = shape;
+    }
+};
+
+class DynamicRankInputShape : public testing::WithParamInterface<std::tuple<InputType>>,
+                              public testing::Test
+{
+public:
+    InputType input_type;
+
+    static std::string getTestCaseName(testing::TestParamInfo<std::tuple<InputType>> obj)
+    {
+        InputType type;
+        std::ostringstream result;
+        type = std::get<0>(obj.param);
+        result << type;
+        return result.str();
+    }
+
+    void SetUp()
+    {
+        const auto& type = std::get<0>(GetParam());
+        input_type = type;
+    }
+};
+
+TEST_P(BasicParamInference, acos)
+{
+    basic_param_inference<op::Acos>(input_type, input_shape);
+}
+
+TEST_P(IncompatibleInputType, acos)
+{
+    incompatible_input_type<op::Acos>(input_type, input_shape);
+}
+
+TEST_P(DynamicRankInputShape, acos)
+{
+    dynamic_rank_input_shape<op::Acos>(input_type);
+}
+
+INSTANTIATE_TEST_CASE_P(type_prop,
+                        BasicParamInference,
+                        testing::Values(std::make_tuple(InputType{element::f32}, Shape{2, 2}),
+                                        std::make_tuple(InputType{element::i32}, Shape{2, 2}),
+                                        std::make_tuple(InputType{element::i16}, Shape{100, 2, 50}),
+                                        std::make_tuple(InputType{element::u16}, Shape{5, 243, 51}),
+                                        std::make_tuple(InputType{element::f64}, Shape{17, 2, 50}),
+                                        std::make_tuple(InputType{element::i64}, Shape{76, 15, 50}),
+                                        std::make_tuple(InputType{element::f16},
+                                                        Shape{2, 20, 5, 5})),
+                        BasicParamInference::getTestCaseName);
```

Comment:
You can simplify test instantiation with use of testing::Combine.

_const std::vector<InputType> precisions = { InputType::f32, InputType::f16, InputType::i32, InputType::i16 };
const std::vector<Shape> shapes = { Shape{2, 2}, Shape{100, 2, 50}, Shape{5, 243, 51}};
INSTANTIATE_TEST_CASE_P(type_prop,
                        BasicParamInference,
                        testing::Combine(testing::ValuesIn(precisions), testing::ValuesIn(shapes)),
                        BasicParamInference::getTestCaseName);_

Reply:
Great, thanks! I just copied it from the example.